packages/elements/src/rating/index.ts

```diff
@@ -111,54 +171,118 @@ export class Rating extends BasicElement {
   }
 
   /**
-   * Process click event to set the new value
-   * @param {number} index index of star
+   * Handles interactive and aria attribute changes
    * @returns {void}
    */
-  private handleTap (index: number): void {
-    if (!this.interactive) {
+  private interactiveChanged (): void {
+    if (this.interactive) {
+      this.setAttribute('role', 'slider');
+      this.setAttribute('aria-valuemin', `${this.interactiveMinValue}`);
+      this.setAttribute('aria-valuenow', this.value);
+      this.setAttribute('aria-valuemax', this.max);
+      this.tabIndex = 0;
+    }
+    else {
+      this.removeAttribute('role');
+      this.removeAttribute('aria-valuemin');
+      this.removeAttribute('aria-valuenow');
+      this.removeAttribute('aria-valuemax');
+      this.removeAttribute('tabindex');
+    }
+  }
+
+  /**
+   * Handles key input
+   * @param event Key down event object
+   * @returns {void}
+   */
+  protected onKeyDown (event: KeyboardEvent): void {
+    if (event.defaultPrevented || !this.interactive) {
       return;
     }
 
-    const valueIndex = this.maxNumber - index;
-    if (this.valuePrevious !== valueIndex) {
-      this.value = valueIndex.toString();
-      // Dispatch Event when value change
+    // Ignore special keys
+    if (event.shiftKey || event.ctrlKey || event.altKey || event.metaKey) {
+      return;
+    }
+
+    switch (event.key) {
+      case 'Right':
+      case 'Up':
+      case 'ArrowRight':
+      case 'ArrowUp':
+        this.stepUp();
+        break;
+      case 'Left':
+      case 'Down':
+      case 'ArrowLeft':
+      case 'ArrowDown':
+        this.stepDown();
+        break;
+      case 'Home':
+        this.updateValue(this.interactiveMinValue);
+        break;
+      case 'End':
+        this.updateValue(this.max);
+        break;
+      default:
+        return;
+    }
+
+    event.preventDefault();
+  }
+
+  /**
+   * Update value and fired value-changed event
+   * @param value value to updated
+   * @returns {void}
+   */
+  private updateValue (value: number | string): void {
```

Comment:
It would be good to make this function to serve one purpose, that is update and notify value not converting anything at all. I would prefer this value to only accept `string`